File: builder/python/BuilderAdv.py
#!/usr/bin/env python
# coding:utf8

class Product(object):
  def __init__(self, builder):
    self._name = builder._name
    self._describe = builder._describe

  def __repr__(self, *args, **kwargs):
    return 'name: '+self._name+'\ndescribe: '+self._describe

class ConcreteBuilder(object):
  def withName(self, name):
    self._name = name
    return self
  def withDescribe(self, describe):
    self._describe = describe
    return self
  def getsult(self):
    return Product(self)

if __name__ == '__main__':
  product = ConcreteBuilder()\
    .withName('product')\
    .withDescribe('Product instance has been created')\
    .getsult()
  print(product)

File: builder/java/BuilderAdv.java
class ProductAdv {
  private String describe;
  private String name;
  ProductAdv(ConcreteBuilderAdv builder) {
    this.describe = builder.getName();
    this.name = builder.getDescribe();
  }

  @Override
  public String toString() {
    return "ProductAdv{" +
        "describe='" + describe + '\'' +
        ", name='" + name + '\'' +
        '}';
  }
}

class ConcreteBuilderAdv {
  private String describe;
  private String name;
  ConcreteBuilderAdv withName(String name) {
    this.name = name;
    return this;
  }

  ConcreteBuilderAdv withDes(String describe) {
    this.describe = describe;
    return this;
  }

  public String getName() {
    return name;
  }

  public String getDescribe() {
    return describe;
  }

  ProductAdv getResult() {
    return new ProductAdv(this);
  }
}

public class BuilderAdv {
  public static void main(String[] args) {
    ProductAdv product = new ConcreteBuilderAdv()
        .withName("productAdv")
        .withDes("ProductAdv instance has been created")
        .getResult();
    System.out.println(product);
  }
}
